rllib/algorithms/algorithm_config.py

```diff
@@ -792,6 +792,16 @@ def validate(self) -> None:
         # not).
         if self._disable_action_flattening is True:
             self.model["_disable_action_flattening"] = True
+        if self.model.get("custom_preprocessor"):
+            deprecation_warning(
+                old="model_config['custom_preprocessor']",
+                help="Custom preprocessors are deprecated, "
+                "since they sometimes conflict with the built-in "
+                "preprocessors for handling complex observation spaces. "
+                "Please use wrapper classes around your environment "
+                "instead.",
+                error=True,
+            )
```

Comment:
As discussed offline, in the long term, we'd like to see preprocessors migrated to connectors.
We still supported custom_preprocessors so far.
The new Catalog does away with that and we can begin erroring out.